src/main/java/org/jenkinsci/plugins/github_branch_source/JwtHelper.java

```diff
@@ -0,0 +1,82 @@
+package org.jenkinsci.plugins.github_branch_source;
+
+import io.jsonwebtoken.JwtBuilder;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import java.security.GeneralSecurityException;
+import java.security.Key;
+import java.security.KeyFactory;
+import java.security.PrivateKey;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.util.Base64;
+import java.util.Date;
+import java.util.Objects;
+
+import static java.util.Objects.requireNonNull;
+
+class JwtHelper {
+
+    /**
+     * Create a JWT for authenticating to GitHub as an app installation
+     * @param githubAppId the app ID
+     * @param privateKey PKC#8 formatted private key
+     * @return JWT for authenticating to GitHub
+     */
+    static String createJWT(String githubAppId, final String privateKey) {
+        requireNonNull(githubAppId, privateKey);
+
+        SignatureAlgorithm signatureAlgorithm = SignatureAlgorithm.RS256;
+
+        long nowMillis = System.currentTimeMillis();
+        Date now = new Date(nowMillis);
+
+        Key signingKey;
+        try {
+            signingKey = getPrivateKeyFromString(privateKey);
+        } catch (GeneralSecurityException e) {
+            throw new IllegalArgumentException("Couldn't parse private key for GitHub app, make sure it's PKCS#8 format", e);
+        }
+
+        JwtBuilder builder = Jwts.builder()
+                .setIssuedAt(now)
+                .setIssuer(githubAppId)
+                .signWith(signingKey, signatureAlgorithm);
+
+        long expMillis = nowMillis + (60 * 1000 * 10);
+        Date exp = new Date(expMillis);
+        builder.setExpiration(exp);
+
+        return builder.compact();
+    }
+
+    /**
+     * Convert a PKCS#8 formatted private key in string format into a java PrivateKey
+     * @param key PCKS#8 string
+     * @return private key
+     * @throws GeneralSecurityException if we couldn't parse the string
+     */
+    private static PrivateKey getPrivateKeyFromString(final String key) throws GeneralSecurityException {
+        if (key.contains("RSA")) {
+            throw new InvalidPrivateKeyException(
+                    "Private key must be a PKCS#8 formatted string, to convert it from PKCS#1 use: "
```

Comment:
we could add a dependency to bouncy castle then we could handle this? or ask users to convert it?

Reply:
Let's start with asking user to do this.  Smoothing can be a later release.

Reply:
There is a [Bouncy Castle API Plugin](https://plugins.jenkins.io/bouncycastle-api/) that was split from Jenkins core in the past and has a wide install base that you can depend on for this kind of thing, but I agree that it seems best to save that for later.